uasc/secure_channel.go

```diff
@@ -485,13 +480,31 @@ func (s *SecureChannel) open(ctx context.Context, instance *channelInstance, req
 		RequestedLifetime:     s.cfg.Lifetime,
 	}
 
-	return s.sendRequestWithTimeout(req, reqID, s.openingInstance, nil, s.cfg.RequestTimeout, func(v interface{}) error {
-		resp, ok := v.(*ua.OpenSecureChannelResponse)
+	ch := make(chan error, 1)
+	err = s.sendAsyncWithTimeout(req, reqID, s.openingInstance, nil, true, s.cfg.RequestTimeout, func(r *response) {
+		if r.Err != nil {
+			ch <- r.Err
+			return
+		}
+		resp, ok := r.V.(*ua.OpenSecureChannelResponse)
 		if !ok {
-			return errors.Errorf("got %T, want OpenSecureChannelResponse", v)
+			ch <- errors.Errorf("got %T, want OpenSecureChannelResponse", r.V)
```

Comment:
Is there a standard status code that we can use in case this bubbles up to the caller?